### ISSUE TITLE
unexpected method entry in debug.Stack() output

### ISSUE THREAD

**Issue by akavel** [golang/go#22508]
Please answer these questions before submitting your issue. Thanks!

### What version of Go are you using (`go version`)?
go version go1.9.2 linux/amd64  

### Does this issue reproduce with the latest release?
yes

### What operating system and processor architecture are you using (`go env`)?
GOARCH="amd64"
GOBIN=""
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/home/akavel/gopath"
GORACE=""
GOROOT="/home/akavel/go"
GOTOOLDIR="/home/akavel/go/pkg/tool/linux_amd64"
GCCGO="gccgo"
CC="gcc"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build776231698=/tmp/go-build -gno-record-gcc-switches"
CXX="g++"
CGO_ENABLED="1"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"

### What did you do?
https://play.golang.org/p/VyVW40dCSU

### What did you expect to see?
```
goroutine 1 [running]:
runtime/debug.Stack(0x31640, 0x252f, 0x8, 0xf0600)
	/usr/local/go/src/runtime/debug/stack.go:24 +0xc0
main.newTracer(0x23c00, 0x252f)
	/tmp/sandbox359837437/main.go:14 +0x20
main.main()
	/tmp/sandbox359837437/main.go:23 +0x20
```

### What did you see instead?
```
goroutine 1 [running]:
runtime/debug.Stack(0x31640, 0x252f, 0x8, 0xf0600)
	/usr/local/go/src/runtime/debug/stack.go:24 +0xc0
main.newTracer(0x23c00, 0x252f)
	/tmp/sandbox359837437/main.go:14 +0x20
main.(*tracer).AddMsg(...)
	/tmp/sandbox359837437/main.go:23
main.main()
	/tmp/sandbox359837437/main.go:23 +0x20
```

Additional notes:
- this is a simplified version of code I have, where this behavior changed since Go 1.5.4 where I last run it (the original code)
- I may be missing something, but it's unclear to me why the AddMsg entry is now present in the stack trace; I'm reporting partially as a means to hopefully learn if there's an explanation if this newer version is more correct than the old one?

**Comment by ALTree** [comment]
This is already fixed on tip (see 7c8a9615c0512815060489a491265c275127b79f)

```
$ gotip version
go version devel +99be9cc02c Tue Oct 31 03:54:16 2017 +0000 linux/amd64

$ gotip run test.go
goroutine 1 [running]:
runtime/debug.Stack(0xc420047f18, 0x40e448, 0x10)
	/home/alberto/go/src/runtime/debug/stack.go:24 +0xa7
main.newTracer(0x4042ec)
	/home/alberto/Desktop/test.go:14 +0x26
main.main()
	/home/alberto/Desktop/test.go:23 +0x26
```

**Comment by akavel** [comment]
Cool, thanks!
